src/base/base_runner.js

```diff
@@ -4,13 +4,55 @@ import shuffle from 'lodash.shuffle';
 import type {
   Logger,
   IRegistry,
+  Hooks,
 } from '../../types';
 
 class BaseRunner {
+  preProcess: () => Promise<void>;
+  healthCheck: () => Promise<void>;
+  terminationCheck: () => Promise<boolean>;
+  errorCount: number;
   registry: IRegistry;
   logger: Logger;
 
-  getActiveSubsciptions(topics: Array < string > = null): Array < string > {
+  constructor(hooks: Hooks = {}) {
+    this.errorCount = 0;
+    this.preProcess = hooks.preProcess || (() => Promise.resolve());
+    this.healthCheck = hooks.healthCheck || (() => Promise.resolve());
+    this.terminationCheck = hooks.terminationCheck || (() => Promise.resolve(false));
+    /* eslint-disable no-console */
+    this.logger = {
+      debug: console.log.bind(console),
+      info: console.log.bind(console),
+      trace: console.log.bind(console),
+      error: console.log.bind(console),
+    };
+    /* eslint-enable no-console */
+  }
+
+  async checks(onFail) {
+    if (await this.terminationCheck()) {
```

Comment:
what will happen if this throws?

Reply:
Good question. Let me see if I can get it set up to not be an issue.